### PR TITLE
[OSCD] Scheduled Task/Job: Cron

### PR DESCRIPTION
**Pull request by alejandroortuno**
Covering:

* Linux Task 10: T1053.003: Cron #1011
* MacOs Task 15: T1053.003: Cron #1012

Regarding the outcome of #1048 we can consider renaming the file name prefix and also move it into the right folder.

### CODE REVIEW

**Comment by yugoslavskiy** [comment]
Hello @alejandroortuno! 

Thank you for your contribution! 
I am not quite sure how to handle this situation with the rule.
But regardless of ongoing discussion, I believe it makes sense to split rule for macOS and Linux. 
Also, I would rather use `processname` and `commandline` as we do in Windows, considering further creating of the `process_creating` category.

**Comment by alejandroortuno** [comment]
Hi @yugoslavskiy, have been thinking on this one and not sure moving into a `processname` / `commandline` will be the best approach though from the keywords option with the full command line. So was thinking an approach using `processname` could be as follows for the following atomic test `echo "* * * * * #{command}" > #{tmp_cron} && crontab #{tmp_cron}` :

```
detection:
  selection_1:
    ProcessName|endswith:
      - '*/echo'
    CommandLine|contains:
      - '"*" > *'
  selection_2:
    ProcessName|endswith:
      - '*/crontab'
```

First thing I noticed is using osquery process_events is the `echo` command does not actually generates a process event. Havent dig the full details although I found this could explain shell functionality does not trigger a process event as it would be the echo binary:

> echo is a shell builtin in Bash and most of the other popular shells like Zsh and Ksh. Its behavior is slightly different from shell to shell. There is also a standalone /usr/bin/echo utility, but usually, the shell built-in version will take precedence. 

Saying so, using the process event we would only be looking at the `crontab` event which might trigger many false positives.

Please let me know your thoughts.

**Comment by yugoslavskiy** [comment]
Hello @alejandroortuno!

> I [...] not sure moving into a processname / commandline will be the best approach though from the keywords option with the full command line.

**Comment by alejandroortuno** [comment]
@yugoslavskiy totally agree, nice detailed explanation. 

> Detect changes in cron jobs that are made via crontab and upload from /tmp. That would be the detection of more like a specific procedure/tool, than behavior. Because crontab could upload tasks from any place, not from /tmp. You can use process_creation category for it (with processname / commandline).

I will focus on this detection though.

**Comment by alejandroortuno** [comment]
@yugoslavskiy got the error on the tests due to the logic of macos / linux sigma rules being the same:

> There are rule files with exactly the same detection logic.

Any thoughts on how we could tackle this as product is different not sure if test should also account for this.

File: rules/linux/macos_schedule_task_job_cron.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+date: 2020/10/06
+references:
+    - https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1053.003/T1053.003.md
+logsource:
+  category: process_creation
+  product: macos
+detection:
+  selection:
+    ProcessName|endswith:
+      - 'crontab'
```
Comment (yugoslavskiy) [inline rules/linux/macos_schedule_task_job_cron.yml:15]: I think that we could bypass the test adding a slash to the processname, this way: `'/crontab'`

Comment (yugoslavskiy) [reply rules/linux/macos_schedule_task_job_cron.yml:15]: Regardless of the test, your rules look perfect. Thank you very much!
I will approve your PR and summon maintainers as soon as you will try to bypass the test, regardless of the result (: 
I belive that we just need to give it a try

Comment (alejandroortuno) [reply rules/linux/macos_schedule_task_job_cron.yml]: just added the slash @yugoslavskiy but seems that the testing phase is having a different issue now:

>  TypeError: 'NoneType' object is not iterable
Makefile:122: recipe for target 'test-sigma2attack' failed
make: *** [test-sigma2attack] Error 1
Error: Process completed with exit code 2.

Comment (yugoslavskiy) [reply rules/linux/macos_schedule_task_job_cron.yml]: It seems that it's just broken. Thank you for giving it a try! And for your contribution (: Well done!

**Review by yugoslavskiy — APPROVED** [review]
@thomaspatzke @Neo23x0 guys, I believe that this PR could be merged (:
please let us know what do you think about it if there is anything you would like to change.

**Pull request merged**